pkg/security/scc/byrestrictions.go

```diff
@@ -111,3 +125,39 @@ func volumePointValue(scc *securityapi.SecurityContextConstraints) points {
 	}
 	return noPoints
 }
+
+// hasCap checks for needle in haystack.
+func hasCap(needle kapi.Capability, haystack []kapi.Capability) bool {
+	for _, c := range haystack {
+		if needle == c {
+			return true
+		}
+	}
+	return false
+}
+
+// capabilitiesPointValue returns a score based on the capabilities allowed,
+// added, or removed by the SCC. This allow us to prefer the more restrictive
+// SCC.
+func capabilitiesPointValue(scc *securityapi.SecurityContextConstraints) points {
+	capsPoints := runCapsDefaultPoints
+	capsPoints += runCapsAddOnePoints * points(len(scc.DefaultAddCapabilities))
+	if hasCap(kapi.CapabilityAll, scc.AllowedCapabilities) {
+		capsPoints += runCapsAllowAllPoints
+	} else if hasCap("ALL", scc.AllowedCapabilities) {
+		capsPoints += runCapsAllowAllPoints
+	} else {
+		capsPoints += runCapsAllowOnePoints * points(len(scc.AllowedCapabilities))
+	}
+	if hasCap("ALL", scc.RequiredDropCapabilities) {
+		capsPoints += runCapsDropAllPoints
+	} else {
+		capsPoints += runCapsDropOnePoints * points(len(scc.RequiredDropCapabilities))
+	}
+	if (capsPoints > runCapsMaxPoints) {
+		return runCapsMaxPoints
+	} else if (capsPoints < runCapsMinPoints) {
+		return runCapsMinPoints
```

Comment:
I'd suggest to remove this logic for max/min caps points. I feel that it makes things more complex but the benefits is too little (if they exist at all).

@pweil- @adelton Objections?